pkg/plugins/mellanox/mellanox_plugin.go

```diff
@@ -99,9 +98,21 @@ func (p *MellanoxPlugin) OnNodeStateChange(old, new *sriovnetworkv1.SriovNetwork
 			attrs.enableSriov = true
 			requireChange = true
 		}
-		if fwCurrent.linkType != EthLinkType && fwCurrent.linkType != PreconfiguredLinkType && fwCurrent.linkType != UknownLinkType {
-			attrs.linkType = EthLinkType
-			requireChange = true
+		if iface.LinkType != "" {
+			if iface.LinkType != EthLinkType && iface.LinkType != InfinibandLinkType {
+				err = fmt.Errorf("mellanox-plugin OnNodeStateChange(): Not supported link type: %s", iface.LinkType)
+				return false, false, err
+			}
+			if fwCurrent.linkType == PreconfiguredLinkType || fwCurrent.linkType == UknownLinkType {
+				err = fmt.Errorf("mellanox-plugin OnNodeStateChange(): Preconfigured NIC or Unkown link type: %s", fwCurrent.linkType)
```

Comment:
I would separate the error in case LinkType is not configurable and Unknown link type.

also a better name would be `FixedLinkType` or `StaticLinkType`. the name `PreConfigured` does not hint in any way that the configuration cannot be changed.

Reply:
I will change it to StaticLinkType